main.py

```diff
@@ -148,5 +172,85 @@ async def keen(context):
     await context.send(result)
 
 
-bot.run(TOKEN)
+@bot.command(name='daily', help='Will give the Daily Leaderboard for specified date')
+async def daily(context, day):
+    # Only respond if used in a channel called 'advent-of-code'
+    if context.channel.name != 'advent-of-code':
+        return
+
+    players = get_players()
+
+    players_day = []
+
+    # Goes through all the players checking if they have data for that day and if they do adding to players_days
+    for player in players:
+        i = 0
+        #This checks they have day data
+        for item in player:
+            print(i, item)
+            i += 1
+        if (i != 5):
+            pass
+        else:
+            if (day in player[4].keys()):
+                players_day.append(player)
+            
+
+    #players_day has all people who have finished one star for that day
+    first_star = []
+    second_star = []
+
+    # Adds all the players which has 
+    for player_day in players_day:
+        if ('1' in player_day[4][day].keys()):
+            
+            first_star.append([player_day[0], int(player_day[4][day]['1']['get_star_ts'])])
+
+        if ('2' in player_day[4][day].keys()):
+            second_star.append([player_day[0], int(player_day[4][day]['2']['get_star_ts'])])
+    
+    #Sorts the two lists on timestamps.(Used insertion as lists aren't too large and easy to change to order on correct data)
+    first_star_sorted = insertion_sort(first_star)
+    second_star_sorted = insertion_sort(second_star)
+
+    print(first_star_sorted)
+    print("\n\n")
+    print(second_star_sorted)
+    print("\n\n")
+    final_table_unsorted = []
+
+    #Adds all the people from first list
+    for i, player in enumerate(first_star_sorted):#
+        final_table_unsorted.append([player[0], (len(players) - i)])
```

Comment:
Should this be a tuple instead of a list? So:
`(player[0], (len(players) - i))`
instead of
`[player[0], (len(players) - i)]`

I think there are a few places throughout where lists are used, but I think tuples would be better, line 233 is another example

Reply:
Should be using tuples throughout now